lib/ar_query_matchers/queries/table_name.rb

```diff
@@ -31,9 +31,8 @@ def active_record_class_for_table(table_name)
         # { 'users' => [User, AtoUser],
         #   'employees => [Employee, PandaFlows::StateFields] }
 
-        # Of all the models that share the same table name sort them by their
-        # relative ancestry and pick the one that all the rest inherit from
-        tables[table_name].min_by { |a, b| a.ancestors.include?(b) }
+        # Pick the class with the shortest distance to the table name
+        tables[table_name].min_by { |a| DidYouMean::Levenshtein.distance(table_name, a.to_s) }
```

Comment:
I'd probably have this pick the class where `table_name.classify == a.to_s` if it exists as that's closer to what folks probably expect (since by convention table names are connected to class names by "classifying" the table name).

Reply:
@alexevanczuk how about this compromise:

```
tables[table_name].min_by { |a| DidYouMean::Levenshtein.distance(table_name.classify, a.to_s) }
```

This way, if there's a direct match, it will be picked, otherwise the shortest distance will match.

Reply:
Sure!